File: public_web/tests/vixrex-validator-parity.test.ts
import { describe, expect, it } from "vitest";
import fixtures from "../../shared/vixrex_validator_parity_fixtures.json";
import { validateField } from "../src/lib/vitrinFieldValidation";

type ValidatorFixture = {
  id: string;
  fieldKey: string;
  input: unknown;
  expectedOk: boolean;
  expectedValue?: unknown;
};

const validatorFixtures = (fixtures as { validator: ValidatorFixture[] }).validator;

describe("Vixrex validator shared parity fixtures — Next.js", () => {
  for (const fixture of validatorFixtures) {
    it(fixture.id, () => {
      const result = validateField(fixture.fieldKey, fixture.input);
      expect(result.ok).toBe(fixture.expectedOk);
      if (fixture.expectedOk) {
        expect(result.ok).toBe(true);
        if (result.ok) expect(result.deger).toEqual(fixture.expectedValue);
      }
    });
  }
});

File: test/vixrex_validator_parity_test.dart
import 'dart:convert';
import 'dart:io';

import 'package:flutter_test/flutter_test.dart';
import 'package:vixrex/services/vixrex_nlu/vixrex_field_validator.dart';

void main() {
  final raw =
      jsonDecode(
            File(
              'shared/vixrex_validator_parity_fixtures.json',
            ).readAsStringSync(),
          )
          as Map<String, dynamic>;
  final fixtures = (raw['validator'] as List).cast<Map<String, dynamic>>();

  group('Vixrex validator shared parity fixtures — Flutter', () {
    for (final fixture in fixtures) {
      test(fixture['id'] as String, () {
        final result = VixrexFieldValidator.validateByKey(
          fixture['fieldKey'] as String,
          fixture['input'],
        );

        expect(result.ok, fixture['expectedOk'] as bool);
        if (fixture['expectedOk'] == true) {
          expect(result.normalizedDeger, fixture['expectedValue']);
        }
      });
    }
  });
}
